packages/melody-runtime/src/filters.js

```diff
@@ -961,3 +961,79 @@ export function strtotime(input, now) {
 
     return date.getTime() / 1000;
 }
+
+export function trim(str, charlist, side = 'both') {
+    //  discuss at: http://phpjs.org/functions/trim/
+    // original by: Kevin van Zonneveld (http://kevin.vanzonneveld.net)
+    // improved by: mdsjack (http://www.mdsjack.bo.it)
+    // improved by: Alexander Ermolaev (http://snippets.dzone.com/user/AlexanderErmolaev)
+    // improved by: Kevin van Zonneveld (http://kevin.vanzonneveld.net)
+    // improved by: Steven Levithan (http://blog.stevenlevithan.com)
+    // improved by: Jack
+    //    input by: Erkekjetter
+    //    input by: DxGx
+    // bugfixed by: Onno Marsman
+    //   example 1: trim('    Kevin van Zonneveld    ');
+    //   returns 1: 'Kevin van Zonneveld'
+    //   example 2: trim('Hello World', 'Hdle');
+    //   returns 2: 'o Wor'
+    //   example 3: trim(16, 1);
+    //   returns 3: 6
+
+    // Melody improvement: Sergio Santoro
+    //   unify ltrim and rtrim as per Twig docs
+
+    if (side !== 'both' && side !== 'left' && side !== 'right') {
+        throw new Error(
+            'Filter "trim". Invalid value ' +
+                side +
+                ' for parameter "side". Valid values are "both", "left", "right".'
+        );
+    }
+
+    var whitespace,
+        l = 0,
+        i = 0;
+    str += '';
+
+    if (!charlist) {
+        // default list
+        whitespace =
+            ' \n\r\t\f\x0b\xa0\u2000\u2001\u2002\u2003\u2004\u2005\u2006\u2007\u2008\u2009\u200a\u200b\u2028\u2029\u3000';
+    } else {
+        // preg_quote custom list
+        charlist += '';
+        whitespace = charlist.replace(
+            /([\[\]\(\)\.\?\/\*\{\}\+\$\^\:])/g,
+            '$1'
+        );
+    }
+
+    if (side === 'both' || side === 'left') {
```

Comment:
I'd suggest a different, more efficient implementation:
Walk through the string from the left and right (as needed) but instead of mutating the string on each character you pass, just keep track of the beginning and ending character index. Then, once calculated, do the substring.

That should be a lot faster.

Reply:
Thanks! I'll have a look.